promslog/slog_test.go

```diff
@@ -94,6 +95,41 @@ func getLogEntryLevelCounts(s string, re *regexp.Regexp) map[string]int {
 	return counters
 }
 
+func TestDurationValues(t *testing.T) {
+	var (
+		buf    bytes.Buffer
+		output string
+		dur, _ = time.ParseDuration("1m30s")
+		config = &Config{Writer: &buf}
+	)
+
+	// Test logfmt.
+	logger := New(config)
+	logger.Info("duration testing", slog.Duration("duration_raw", dur), slog.String("duration_string", dur.String()))
+	output = buf.String()
+	// Print logs for humans to see, if needed.
+	fmt.Println(output)
+
+	require.NotContainsf(t, output, "duration_raw=\"90000000000\"", "Expected duration to be output as Go duration string \"1m30s\", got \"90000000000\"")
+
+	buf.Reset()
+
+	// Test json.
```

Comment:
we can use test cases here where the config is customized, I think we can set format there as well...

Reply:
That's fair, I can clean it up a bit :+1: